processor/metricstransformprocessor/config.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstransformprocessor
+
+import "go.opentelemetry.io/collector/config/configmodels"
+
+// Config defines configuration for Resource processor.
+type Config struct {
+	configmodels.ProcessorSettings `mapstructure:",squash"`
+
+	// MetricName is used to select the metric to operate on.
+	// REQUIRED
+	MetricName string `mapstructure:"metric_name"`
+
+	// Action specifies the action performed on the matched metric.
+	// REQUIRED
+	Action ConfigAction `mapstructure:"action"`
+
+	// NewName specifies the name of the new metric when inserting or updating.
+	// REQUIRED only if Action is INSERT.
+	NewName string `mapstructure:"new_name"`
+
+	// Operations contains a list of operations that will be performed on the selected metrics.
+	Operations []Operation `mapstructure:"operations"`
+}
+
+// Operation defines the specific operation performed on the selected metrics.
+type Operation struct {
+	// Action specifies the action performed for this operation.
+	// REQUIRED
+	Action OperationAction `mapstructure:"action"`
```

Comment:
Can we have the action be implied by the fields that are set?. i.e.

```yaml
- rename_labels_to:
   cpu: core
```

or
```yaml
- keep_labels:
  - one
  - two
  aggregation_type: max
```

(that is, fields called rename_labels_to and keep_labels, where if they have a value, that's the operation to perform)

Reply:
Same as [above](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/347#discussion_r444497901)